File: P/746_Min_Cost_Climbing_Stairs.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int[] dp = new int[cost.length + 1];
        int min = 0;
        
        for(int i = 2; i <= cost.length; i++)
            dp[i] = Math.min(cost[i - 2] + dp[i - 2], cost[i - 1] + dp[i - 1]);

        return dp[dp.length - 1];
    }
}

File: 주영/DP/746. Min Cost Climbing Stairs.js
/**
 * @param {number[]} cost
 * @return {number}
 */
var minCostClimbingStairs = function (cost) {
  let dp = [];
  dp[0] = 0;
  dp[1] = 0;

  for (let i = 2; i < cost.length + 1; i++) {
    dp[i] = Math.min(dp[i - 1] + cost[i - 1], dp[i - 2] + cost[i - 2]);
  }

  // console.log(dp[cost.length]);

  return dp[cost.length];
};

let cost = [1, 100, 1, 1, 1, 100, 1, 1, 100, 1];
minCostClimbingStairs(cost);
